etc/notes/CHANGES_6.0.0.md

```diff
@@ -2,23 +2,339 @@
+
+The associated options interface with this API has also been removed: `AddUserOptions`.
+
+See `createUser` documentation: https://www.mongodb.com/docs/manual/reference/command/createUser/
+
+```ts
+const db = client.db('admin');
+// Example addUser usage
+await db.addUser('myUsername', 'myPassword', { roles: [{ role: 'readWrite', db: 'mflix' }] });
+// Example equivalent command usage
+await db.command({
+  createUser: 'myUsername',
+  pwd: 'myPassword',
+  roles: [{ role: 'readWrite', db: 'mflix' }]
+});
+```
+
+### `collection.stats()` removed
+
+The `collStats` command is deprecated starting in server v6.2 so the driver is removing its bespoke helper in this major release. The `collStats` command is still available to run manually via `await db.command()`. However, the recommended migration is to use the `$collStats` aggregation stage. You can find more information here: [$collStats](https://www.mongodb.com/docs/upcoming/reference/operator/aggregation/collStats/#mongodb-pipeline-pipe.-collStats).
+
+The following interfaces associated with this API have also been removed: `CollStatsOptions` and `WiredTigerData`.
+
+### `BulkWriteResult` deprecated properties removed
+
+The following deprecated properties have been removed as they duplicated the functionality provided by the common driver spec compliant properties. The list indicates what properties provide the correct migration:
+
+- `BulkWriteResult.nInserted` -> `BulkWriteResult.insertedCount`
+- `BulkWriteResult.nUpserted` -> `BulkWriteResult.upsertedCount`
+- `BulkWriteResult.nMatched` -> `BulkWriteResult.matchedCount`
+- `BulkWriteResult.nModified` -> `BulkWriteResult.modifiedCount`
+- `BulkWriteResult.nRemoved` -> `BulkWriteResult.deletedCount`
+- `BulkWriteResult.getUpsertedIds` -> `BulkWriteResult.upsertedIds` / `BulkWriteResult.getUpsertedIdAt(index: number)`
+- `BulkWriteResult.getInsertedIds` -> `BulkWriteResult.insertedIds`
+
+### Deprecated SSL options have been removed
+
+The following options have been removed with their supported counterparts listed after the ->
+
+- `sslCA` -> `tlsCAFile`
+- `sslCRL` -> `tlsCRLFile`
+- `sslCert` -> `tlsCertificateKeyFile`
+- `sslKey` -> `tlsCertificateKeyFile`
+- `sslPass` -> `tlsCertificateKeyFilePassword`
+- `sslValidate` -> `tlsAllowInvalidCertificates`
+- `tlsCertificateFile` -> `tlsCertificateKeyFile`
+
+### The deprecated `keepAlive` and `keepAliveInitialDelay` options have been removed
+
+TCP keep alive will always be on and now set to a value of 30000ms.
+
+## 🪦 Removal of "dead" code
+
+The removed functionality listed in this section was either unused or not useful outside the driver internals.
+
+### `MongoError` and its subclasses now have internal constructors
```

Comment:
```suggestion
### `MongoError` and its subclasses are no longer covered under semantic versioning
```

Do we we think this still fits under dead code removal, or would a separate heading be more appropriate?

Reply:
I don't think this is the best title for this change. We are trying to indicate that something that the users should not have been using to begin with is now more clearly marked as such. Maybe something like "Constructors for `MongoError` and its subclasses now clearly indicate that they are meant for internal use only."